realm/realm-library/src/main/java/io/realm/RealmMigration.java

```diff
@@ -17,49 +17,55 @@
 package io.realm;
 
 /**
- * The RealmMigration class is used to describe the migration of one Realm schema to another.
+ * The RealmMigration class is used to perform the migration of one Realm schema to another.
  * The schema for a Realm is defined by all classes in a project that extend {@link io.realm.RealmObject}, so any
  * changes to these classes will require a migration.
  *
  * To support migrations from any previous schemaVersion to the newest, the following pattern is recommended when
  * writing a migration:
  *
- * <pre>
+ * {@code
  * public class CustomMigration implements RealmMigration {
  *   \@Override
- *   public long execute(Realm realm, long schemaVersion) {
- *     if (schemaVersion == 0) {
+ *   public long migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+ *     RealmSchema schema = realm.getSchema();
+ *
+ *     if (oldVersion == 0) {
  *       // Migrate from v0 to v1
- *       schemaVersion++;
+ *       oldVersion++;
  *     }
  *
- *     if (schemaVersion == 0) {
- *       // Migrate from v0 to v1
- *       schemaVersion++;
+ *     if (oldVersion == 1) {
+ *       // Migrate from v1 to v2
+ *       oldVersion++;
  *     }
  *
- *     return schemaVersion;
+ *     if (oldVersion < newVersion) {
+ *         throw new IllegalStateException(String.format("Migration missing from v%d to v%d", oldVersion, newVersion));
+ *     }
  *   }
  * }
- * </pre>
+ * }
  *
- * During development when model classes can change frequently, it is possible to use
+ * During development when RealmObject classes can change frequently, it is possible to use
  * {@link io.realm.Realm#deleteRealm(RealmConfiguration)}. This will delete the database file and eliminate the need for
  * any migrations.
  *
- * @see Realm#migrateRealm(RealmConfiguration)
- * @see Realm#migrateRealm(RealmConfiguration, RealmMigration)
+ * @see io.realm.RealmConfiguration.Builder#schemaVersion(long)
+ * @see io.realm.RealmConfiguration.Builder#migration(RealmMigration)
+ * @see io.realm.RealmConfiguration.Builder#deleteRealmIfMigrationNeeded()
  */
 public interface RealmMigration {
 
```

Comment:
How about adding `long getSchemaVersion()` here and remove(deprecate) `RealmConfiguration.Builder#schemaVersion(long)` since Migration logic and schema version are tightly coupled.